spec.html

```diff
@@ -47025,12 +47107,105 @@ <h1>
+              <td>[[Value]]</td>
+              <td>an ECMAScript language value</td>
+              <td>The value produced by evaluation of [[ParseNode]].</td>
+            </tr>
+            <tr>
+              <td>[[Elements]]</td>
+              <td>a List of JSON Parse Records</td>
+              <td>JSON Parse Records corresponding with the elements of a [[Value]] that is an Array, in order. If [[Value]] is not an Array, the List will be empty.</td>
+            </tr>
+            <tr>
+              <td>[[Entries]]</td>
+              <td>a List of JSON Parse Records</td>
+              <td>JSON Parse Records corresponding with the entries of a [[Value]] that is a non-Array Object, in source order. If [[Value]] is not a non-Array Object, the List will be empty.</td>
+            </tr>
+          </table>
+        </emu-table>
+      </emu-clause>
+
+      <emu-clause id="sec-createjsonparserecord" type="abstract operation">
+        <h1>
+          CreateJSONParseRecord (
+            _parseNode_: a Parse Node,
+            _key_: a property name,
+            _val_: an ECMAScript language value,
+          ): a JSON Parse Record
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It recursively combines a _parseNode_ parsed from JSON text and the _val_ produced by its evaluation.</dd>
+        </dl>
+        <emu-alg>
+          1. Let _typedValNode_ be ShallowestContainedJSONValue of _parseNode_.
+          1. Assert: _typedValNode_ is not ~empty~.
+          1. Let _elements_ be a new empty List.
+          1. Let _entries_ be a new empty List.
+          1. If _val_ is an Object, then
+            1. Let _isArray_ be ! <emu-meta suppress-effects="user-code">IsArray(_val_)</emu-meta>.
+            1. If _isArray_ is *true*, then
+              1. Assert: _typedValNode_ is an |ArrayLiteral| Parse Node.
+              1. Let _contentNodes_ be ArrayLiteralContentNodes of _typedValNode_.
+              1. Let _len_ be the number of elements in _contentNodes_.
+              1. Let _valLen_ be ! <emu-meta suppress-effects="user-code">LengthOfArrayLike(_val_)</emu-meta>.
+              1. Assert: _valLen_ = _len_.
+              1. Let _I_ be 0.
+              1. Repeat, while _I_ &lt; _len_,
+                1. Let _propName_ be ! ToString(𝔽(_I_)).
+                1. Let _elementParseRecord_ be CreateJSONParseRecord(_contentNodes_[_I_], _propName_, ! <emu-meta suppress-effects="user-code">Get(_val_, _propName_)</emu-meta>).
+                1. Append _elementParseRecord_ to _elements_.
+                1. Set _I_ to _I_ + 1.
+            1. Else,
+              1. Assert: _typedValNode_ is an |ObjectLiteral| Parse Node.
+              1. Let _propertyNodes_ be PropertyDefinitionNodes of _typedValNode_.
+              1. NOTE: Because _val_ was produced from JSON text and has not been modified, all of its property keys are Strings and will be exhaustively enumerated in source text order.
```

Comment:
Is this true? In `Object.keys({ "a": 1, "0": 2 })` they are not enumerated in source text order.